### ISSUE TITLE
Integrate OpenBSD's pledge into Kovri

### ISSUE THREAD

**Issue by stoffu** [monero-project/kovri#657]
---
**By submitting this issue, I confirm the following:**

- I have read and understood the [contributor guide](https://github.com/anonimal/kovri-docs/blob/master/developer/CONTRIBUTING.md).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.
---

I'm taking a Reddit post by u/garlicgambit to here since it seems to be forgotten. Not that I know anything about it :)

https://www.reddit.com/r/Kovri/comments/6bdk0a/integrate_openbsds_pledge_into_kovri/

> Are you familiar with OpenBSD's pledge? Link:
http://man.openbsd.org/pledge.2
Is this something you are willing to consider to integrate into Kovri? I'm asking because Kovri is still a relatively young project and integrating it at this stage would probably be easier then doing it later on. Pledge could help discover bugs in general and perhaps stop some exploits on OpenBSD.
I know you guys are busy, but I thought I'd ask. Thanks for creating this important project.

**Comment by anonimal** [comment]
Thank you, @stoffu!

**Comment by anonimal** [comment]
I've gained reassurances from fellow developers at DefCon25 when conferring about pledge and, after a final review of the [documentaion](https://man.openbsd.org/pledge.2) and [proposal](http://www.openbsd.org/papers/hackfest2015-pledge/index.html), and given the lack of C++ pledge support (especially in our dependencies), and given the very little amount of low-level calls we *do* implement:  I've concluded that pledge is not worth pursuing for our codebase.

I do believe we *could* pledge a few of the calls in both the *nix-based daemon impl and MTU (testing needed for both) but, aside from that, we simply don't have an interface available for a global refactor nor would committing such a refactor be cost-effective for the "security" trade-off (IMHO, proper opsec/sandboxing/RBAC/ACL **would be more effective** for the end-user and would minimize our load).

I think pledge/cap-based control is an important practice when used in *appropriate use-cases*. We should close this issue pending any affirmative feedback or if any related PR's do come in re: the daemon + MTU.
